### ISSUE TITLE
DrawingSurface::DrawSurface doesn't process transparency properly

### ISSUE THREAD

**Issue by fernewelten** [adventuregamestudio/ags#1513]
**Describe the bug**
In a true-colour game, when a DrawingSurface is drawn onto a DrawingSurface, the calculations don't process the transparency parameter and/or the alpha channel of the drawing surfaces properly. See screenshots below.

Also, apparently transparency cannot be `100`. This should be an allowed value and result in an unchanged surface (because something is drawn onto it that is completely transparent).

**AGS Version**
Build 3.5.1.14
__Edit:__ A quick test seems to show that AGS4 __does__ handle `DrawingSurface::DrawSurface()` properly.

**Game**
[Test351.zip](https://github.com/adventuregamestudio/ags/files/7920681/Test351.zip)

Here's the relevant code of Room 1:
```
DynamicSprite *ds;

function test(int transp)
{
    DrawingSurface *dsu = ds.GetDrawingSurface();
    dsu.DrawingColor = Game.GetColorFromRGB(255, 255, 255);
    dsu.DrawRectangle(10, 10, 50, 50);
    DrawingSurface *dsu_copy = dsu.CreateCopy();
    dsu.DrawingColor = Game.GetColorFromRGB(250, 100, 100);
    dsu.DrawCircle(20, 20, 20);
    dsu.DrawSurface(dsu_copy, transp);
    dsu.Release();
    dsu_copy.Release();
    player.Say("This is DrawSurface with %d %% transparency", transp);
}

function room_AfterFadeIn()
{
    ds = DynamicSprite.Create(50, 50, true);
    oTestTest.Graphic = ds.Graphic;
    
    test(99);
    test(70);
    test(20);
    test(1);
    test(0);
}
```

The setup in Room 1:
![image](https://user-images.githubusercontent.com/41213694/150683915-21a5cc28-df10-4fc2-b0c3-f97d372ddbfb.png)

**To Reproduce**
Run game, observe screens:

**Screenshots and expected behaviour**
![image](https://user-images.githubusercontent.com/41213694/150683985-6ba07571-cb9b-499b-aa4b-69c56cf11297.png)
The square seems to be completely cut out of the circle although it is nearly completely transparent as per the `transparency` parameter (99 %).
I'd expect some of the circle to show through.

![image](https://user-images.githubusercontent.com/41213694/150684064-ec1ec5bd-aeac-49f3-8bcc-32288042285a.png)
The circle has completely disappeared. 
Outside of the white square, the alpha channel of the copy should be "transparent", so copying should not obliterate the circle in that area.

![image](https://user-images.githubusercontent.com/41213694/150684209-3300516a-12db-41aa-9938-04288c59a659.png)
Transparency 20 % seems to do exactly the same as transparency 70 %. That can't be right.

![image](https://user-images.githubusercontent.com/41213694/150684265-0b333f24-de79-4970-b2da-ded5801b8ca3.png)
Even transparency 1 % seems to do exactly the same as transparency 70 %

![image](https://user-images.githubusercontent.com/41213694/150684314-ed2dbcee-46ff-4cfe-8671-65a89f1f04bb.png)
At 0 % transparency, the copy seems to replace the surface completely.
But, seeing that most of the copy has a completely transparent alpha channel, I'd expect the red circle to stay on surface wherever the white square isn't.

**Desktop (please complete the following information):**
 - Windows 10, German edition

**Smartphone (please complete the following information):**
- n/a

**Comment by ericoporto** [comment]
Curiously the `Test351.ags` game in `Compiled/Data` did not ran for me. I rebuilt the attached game project and ran it with the latest 3.6.0.X editor release. And the bug doesn't happen in the master, so I tried 3.5.1.14, and it happened there.

**Comment by ivan-mogilko** [comment]
I recall merging DrawingImage and DrawingSurface internal implementation to go through the same draw function in the end, so maybe that actually fixed that.

**Comment by ivan-mogilko** [comment]
Fixed in release-3.5.1 with 2710b999c8af56efbddccee69635ac23fd52b657.

Don't know if/when there will be a next 3.5.1 yet, I'd rather wait for a while in case someone finds another bug there.

**Comment by fernewelten** [comment]
> I wonder when it got broken though.

I seem to have unearthed a function that hasn't been used for a long time, or else somebody would have complained already. 😀
